src/pages/Auth/LoginPage/hooks/useLogin.ts

```diff
@@ -0,0 +1,74 @@
+import { isKoinError, sendClientError } from '@bcsdlab/koin';
+import { sha256 } from '@bcsdlab/utils';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { auth } from 'api';
+import { LoginResponse } from 'api/auth/entity';
+import { useLoginRedirect } from 'utils/hooks/auth/useLoginRedirect';
+import { setCookie } from 'utils/ts/cookie';
+import showToast from 'utils/ts/showToast';
+import { useTokenStore } from 'utils/zustand/auth';
+
+interface IsAutoLogin {
+  isAutoLoginFlag: boolean;
+}
+
+interface UserInfo {
+  userId: string;
+  password: string;
+}
+
+const emailLocalPartRegex = /^[a-z_0-9]{1,12}$/;
+
+export const useLogin = (state: IsAutoLogin) => {
+  const { setToken, setRefreshToken } = useTokenStore();
+  const { redirectAfterLogin } = useLoginRedirect();
+  const queryClient = useQueryClient();
+
+  const postLogin = useMutation({
+    mutationFn: auth.login,
+    onSuccess: (data: LoginResponse) => {
+      if (state.isAutoLoginFlag) {
+        setRefreshToken(data.refresh_token);
+      }
+
+      queryClient.invalidateQueries();
+      setCookie('AUTH_TOKEN_KEY', data.token);
+      setToken(data.token);
+      redirectAfterLogin();
+    },
+    onError: (error) => {
+      if (isKoinError(error)) {
+        showToast('error', error.message || '로그인에 실패했습니다.');
+      } else {
+        sendClientError(error);
+        showToast('error', '로그인에 실패했습니다.');
+      }
+    },
+  });
+
+  const login = async (userInfo: UserInfo) => {
+    if (userInfo.userId === '') {
+      showToast('error', '계정을 입력해주세요');
+      return;
+    }
+    if (userInfo.password === '') {
+      showToast('error', '비밀번호를 입력해주세요');
+      return;
+    }
+    if (userInfo.userId.indexOf('@koreatech.ac.kr') !== -1) {
+      showToast('error', '계정명은 @koreatech.ac.kr을 빼고 입력해주세요.');
+      return;
+    }
+    if (!emailLocalPartRegex.test(userInfo.userId)) {
+      showToast('error', '아우누리 계정 형식이 아닙니다.');
+      return;
+    }
+    const hashedPassword = await sha256(userInfo.password);
+
+    postLogin.mutate({
+      email: `${userInfo.userId}@koreatech.ac.kr`,
+      password: hashedPassword,
+    });
+  };
+  return login;
```

Comment:
훅 정렬 방식이나 중간 개행을 할 때 자신만의 기준이 있을까요?
코드를 보는데 아직 그 기준이 무엇인지 잘 모르겠어서, 제 취향껏 띄워주고 싶어서 남겨봅니다. ㅎㅎ

```suggestion
  };
  
  return login;
```

Reply:
lint에 주로 의존하긴 하지만... 저도 return문 이전에 개행하는 것을 선호하긴 합니다.

제 기준을 나름대로 적어보면
주로 의존성이 있는 코드끼리 뭉쳐서 작성하고, 의존성이 낮은 코드는 따로 개행을 줘서 가독성을 높이려고 하는 편인 것 같아요!
그리고 상태와 함수, 조건문, 요청, 리턴문 등 각각 뭉쳐서 작성하고 구분하기 위해 개행을 주는 편입니다.